src/hidimstat/ensemble_clustered_inference.py

```diff
@@ -1,45 +1,328 @@
+    )
+
+    beta_hat = _rescale_beta(beta_hat, n_features=pval.shape[0], ward=ward)
+
+    return beta_hat, pval, pval_corr, one_minus_pval, one_minus_pval_corr
+
+
+def clustered_inference(
+    X_init,
+    y,
+    ward,
+    n_clusters,
+    scaler_sampling=None,
+    train_size=1.0,
+    groups=None,
+    seed=0,
+    n_jobs=1,
+    verbose=1,
+    **kwargs,
 ):
+    """
+    Clustered inference algorithm for statistical analysis of
+    high-dimensional data.
 
-    beta_hat = np.asarray(list_beta_hat)
-    list_pval = np.asarray(list_pval)
-    list_pval_corr = np.asarray(list_pval_corr)
-    list_one_minus_pval = np.asarray(list_one_minus_pval)
-    list_one_minus_pval_corr = np.asarray(list_one_minus_pval_corr)
+    This algorithm implements the method described in :cite:`chevalier2022spatially` for
+    performing statistical inference on high-dimensional linear models
+    using feature clustering to reduce dimensionality.
 
-    beta_hat = np.mean(list_beta_hat, axis=0)
+    Parameters
+    ----------
+    X_init : ndarray, shape (n_samples, n_features)
+        Original high-dimensional input data matrix.
+
+    y : ndarray, shape (n_samples,) or (n_samples, n_times)
+        Target variable(s). Can be univariate or multivariate (temporal) data.
+
+    ward : sklearn.cluster.FeatureAgglomeration
+        Hierarchical clustering object that implements Ward's method for
+        feature agglomeration.
+
+    n_clusters : int
+        Number of clusters to use for dimensionality reduction.
+
+    scaler_sampling : sklearn.preprocessing object, optional (default=None)
+        Scaler to standardize the clustered features.
+
+    train_size : float, optional (default=1.0)
+        Fraction of samples to use for computing the clustering.
+        When train_size=1.0, all samples are used.
+
+    groups : ndarray, shape (n_samples,), optional (default=None)
+        Sample group labels for stratified subsampling.
+
+    seed : int, optional (default=0)
+        Random seed for reproducible subsampling.
+
+    n_jobs : int, optional (default=1)
+        Number of parallel jobs for computation.
+
+    verbose : int, optional (default=1)
+        Verbosity level for progress messages.
+
+    **kwargs : dict
+        Additional arguments passed to the statistical inference function.
 
-    if method == "quantiles":
+    Returns
+    -------
+    ward_ : FeatureAgglomeration
+        Fitted clustering object.
+
+    beta_hat : ndarray, shape (n_clusters,) or (n_clusters, n_times)
+        Estimated coefficients at cluster level.
 
-        pval = aggregate_quantiles(list_pval, gamma_min)
-        pval_corr = aggregate_quantiles(list_pval_corr, gamma_min)
-        one_minus_pval = aggregate_quantiles(list_one_minus_pval, gamma_min)
-        one_minus_pval_corr = aggregate_quantiles(list_one_minus_pval_corr, gamma_min)
+    theta_hat : ndarray
+        Estimated precision matrix.
 
-    elif method == "medians":
+    omega_diag : ndarray
```

Comment:
```suggestion
    precision_diag : ndarray
```

Reply:
I'm confused. Is this the covariance or precision ?

Reply:
It's the output of desparsified Lasso. Let's talk about it in the PR #127 for the moment.